### ISSUE TITLE
Bug: consistent-type-definitions leaves trailing parenthesis when converting a paren-wrapped type to an interface

### ISSUE THREAD

**Issue by rdecoito** [typescript-eslint/typescript-eslint#10233]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Issue Description

This isn't actually a complex issue, but I didn't immediately see how to set eslint version 9 in the playground and I wasn't interested in learning, since I already had a reproduction repo set up.

The issue is that when you use the --fix argument with the `@typescript-eslint/consistent-type-definitions` rule, it will leave a trailing parenthesis if the type it converts was wrapped in parens.

For example, the following code

```typescript
export type Interface = ({
    [key: string]: unknown;
});
```

becomes

```typescript
export interface Interface {
    [key: string]: unknown;
});
```

triggering a syntax error.

I tried this on the latest version of typescript-eslint, including 8.12.2 and 8.12.3-alpha.1

### Reproduction Repository Link

https://github.com/rdecoito/repro-consistent-type-definitions

### Repro Steps

Copy-pasted from README in repo:

1. Clone this repo
2. Run `npm i`
3. Run `npm run lint`
	- You'll see an error reported that the type defined in src/index.ts should be an interface instead
4. Run `npm run lint -- --fix`
	- You'll see a new syntax error reported. Open src/index.ts to see the problem: the rule left behind a parenthesis that it shouldn't have

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `typescript-eslint`             | `~8.11.0` |
| `TypeScript`                           | `5.6.3` |
| `ESLint`                               | `~9.13.0` |
| `node`                                 | `20.18.0` |

**Comment by kirkwaiblinger** [comment]
[playground link](https://typescript-eslint.io/play/#ts=5.4.5&showAST=types&fileType=.tsx&code=KYDwDg9gTgLgBDAnmYcCSA7GwoDMCGAxqgLxwAUA3gFBxwDaA1sIgFxwDOMUAlhgOYBddgFcMjDBADuGANzUAvgEpZQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6MgeyeUuX0Ra0ipWgBNEAM2aV8lLqgyQA7gENoTSODABfENqA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false).

(Please do try it next time! it really is easy! I just copied your code in and added `consistent-type-definitions` to the rules object).
